base/sort.jl

```diff
@@ -349,6 +359,38 @@ function sort!(v::AbstractVector, lo::Int, hi::Int, a::MergeSortAlg, o::Ordering
     return v
 end
 
+function sort!(v::AbstractVector, lo::Int, hi::Int, a::PartialQuickSort,
+               o::Ordering)
+    k = a.k
+    while lo < hi
+        hi-lo <= SMALL_THRESHOLD && return sort!(v, lo, hi, SMALL_ALGORITHM, o)
+        pivot = v[(lo+hi)>>>1]
+        i, j = lo, hi
+        while true
+            while lt(o, v[i], pivot); i += 1; end
+            while lt(o, pivot, v[j]); j -= 1; end
+            i <= j || break
+            v[i], v[j] = v[j], v[i]
+            i += 1; j -= 1
+        end
+        if lo < j
+            if j - lo <= k
+                sort!(v, lo, j, QuickSort, o)
+            else
+                sort!(v, lo, j, PartialQuickSort(k), o)
+            end
```

Comment:
If `k` is changing (as in this version of the code), choosing between `QuickSort` and `PartialQuickSort` makes some sense (since `PartialQuickSort` does more work), but if you leave `k` constant, the amount of extra work is trivial, and it simplifies the code to simply always call `PartialQuickSort`.